tests/gui/test_helper_find.py

```diff
@@ -0,0 +1,145 @@
+# ###################################################
+# Copyright (C) 2008-2016 The Unknown Horizons Team
+# team@unknown-horizons.org
+# This file is part of Unknown Horizons.
+#
+# Unknown Horizons is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the
+# Free Software Foundation, Inc.,
+# 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+# ###################################################
+
+import unittest
+
+import mock
+
+from tests.gui.helper import GuiHelper
+
+
+class Widget(object):
+	"""
+	Fake widget with a name, children and possible a parent. All children of this widget will
+	get their parent attribute set to this widget.
+	"""
+	def __init__(self, name, children=None):
+		self.name = name
+		self.parent = None
+		self.children = children or []
+		for c in self.children:
+			c.parent = self
+
+	def __hash__(self):
+		return id(self)
+
+
+class FindWidgetTest(unittest.TestCase):
+	"""
+	Tests about finding widgets in pychan's internal lists.
+	"""
+	def setUp(self):
+		self.widgets = set()
+
+		def w(name, children=None):
+			widget = Widget(name, children)
+			self.widgets.add(widget)
+			return widget
```

Comment:
You're doing it explicitly for every widget, because you also have to create them through this function. Got it.